rust_src/src/dired_unix.rs

```diff
@@ -95,6 +100,270 @@ impl LispObjectExt for LispObject {
     }
 }
 
+struct DirFiles {
+    directory: String,
+    full: bool,
+    match_re: LispObject,
+    nosort: bool,
+    attrs: bool,
```

Comment:
I don't think I'm alone in disliking inverted booleans, or boolean arguments. These could easily be enums.

Reply:
Sounds good. Lemme give it a shot.